wdl/model/draft3/src/main/scala/wdl/model/draft3/elements/ScatterElement.scala

```diff
@@ -3,4 +3,21 @@ package wdl.model.draft3.elements
 final case class ScatterElement(scatterName: String,
                                 scatterExpression: ExpressionElement,
                                 scatterVariableName: String,
-                                graphElements: Seq[WorkflowGraphElement]) extends WorkflowGraphElement
+                                graphElements: Seq[WorkflowGraphElement]) extends WorkflowGraphElement {
+
+  // Scatter names do not contain intrinsic information about the scatter; rather they are a sort
+  // of hash based on the declarations's physical location in the source.
+  override def equals(other: scala.Any): Boolean = {
+    other match {
+      case otherScatter: ScatterElement =>
+        this.scatterExpression == otherScatter.scatterExpression &&
```

Comment:
I forget the exact interplay between `.equals` and `==` in scala but since you're overriding `equals`, does it make sense to use `.equals` in these sub-comparisons?

Reply:
Wasn't sure so [looked it up](https://stackoverflow.com/a/7681243/818054):
>You normally use `==`, it routes to `equals`, except that it treats `null`s properly. Reference equality (rarely used) is `eq`.

So we could do so on a style basis only, which I'm 🤷‍♂️ on

Reply:
oh, it's `eq` that's different...

Ok, so, never mind, looks good to me